src/wizards/validators.ts

```diff
@@ -0,0 +1,76 @@
+// ------------------ Cluster validators
+
+export async function validateBroker(broker: string): Promise<string | undefined> {
+    return validateFieldRequired('Broker', broker);
+}
+
+export async function validateClusterName(cluster: string, existingClusterNames: string[]): Promise<string | undefined> {
+    const result = validateFieldRequired('Cluster name', cluster);
+    if (result) {
+        return result;
+    }
+    return validateFieldUniqueValue('Cluster name', cluster, existingClusterNames);
+}
+
+export async function validateAuthentificationUserName(userName: string): Promise<string | undefined> {
+    return validateFieldRequired('User name', userName);
+}
+
+// ------------------ Topic validators
+
+const maxNameLength = 249
+const legalChars = /^[a-zA-Z0-9\\.\\-]*$/;
+
+export async function validateTopicName(topic: string, existingTopicNames: string[]): Promise<string | undefined> {
+    // See topic name validation rule at https://github.com/apache/kafka/blob/8007211cc982d8458223e866c1ee7d94b69e0249/core/src/main/scala/kafka/common/Topic.scala#L33
+    const result = validateFieldRequired('Topic', topic);
+    if (result) {
+        return result;
+    }
+    else if (topic == "." || topic == "..") {
+        return "Topic cannot be \".\" or \"..\"";
+    }
+    else if (topic.length > maxNameLength) {
+        return `Topic is illegal, can't be longer than ${maxNameLength} characters`;
+    }
+    else if (!legalChars.test(topic)) {
+        return `Topic ${topic} is illegal, contains a character other than ASCII alphanumerics, '.', '_' and '-'`;
+    }
+    return validateFieldUniqueValue('Topic', topic, existingTopicNames);
+}
+
+export async function validatePartitions(partitions: string): Promise<string | undefined> {
+    const result = validateFieldRequired('Partitions', partitions);
+    if (result) {
+        return result;
+    }
+    return validateFieldPositiveNumber('Partitions', partitions);
+}

```

Comment:
We should prolly limit the max value for the number of topic partitions. Looks like 10 seems to be the max recommended amount, So setting [100](https://medium.com/swlh/choosing-right-partition-count-replication-factor-apache-kafka-cf50b1bc75cf) as a limit might be fine. @jlandersen WDYT?

Kafka clusters seem limited to [200K partions overall](https://blogs.apache.org/kafka/entry/apache-kafka-supports-more-partitions#:~:text=Apache%20Kafka%20Supports%20200K%20Partitions%20Per%20Cluster%20%3A%20Kafka), if anything that should be the hard limit.